modules/git/blob.go

```diff
@@ -21,15 +20,17 @@ func (b *Blob) Name() string {
 	return b.name
 }
 
-// GetBlobContent Gets the content of the blob as raw text
-// The max content size depends on MAX_DISPLAY_FILE_SIZE in app.ini
-func (b *Blob) GetBlobContent() (string, error) {
+// GetBlobContent Gets the limited content of the blob as raw text
+func (b *Blob) GetBlobContent(limit int64) (string, error) {
+	if limit <= 0 {
+		return "", nil
+	}
 	dataRc, err := b.DataAsync()
 	if err != nil {
 		return "", err
 	}
 	defer dataRc.Close()
-	buf := make([]byte, setting.UI.MaxDisplayFileSize)
+	buf := make([]byte, limit)
```

Comment:
Here is a nit: since default MaxDisplayFileSize is 8M, so each `GetBlobContent` allocates 8M memory, but in most cases it only reads hundreds or thousands of bytes. It wastes memory. But it can be fixed later.

Reply:
`if limit > b.size` then `limit = b.size`?

Reply:
Done in 6d950bfed961ca79356347f35cc3c580aa8455cc